source/extensions/filters/http/ip_tagging/ip_tagging_filter.cc

```diff
@@ -112,6 +108,46 @@ void IpTaggingFilter::setDecoderFilterCallbacks(Http::StreamDecoderFilterCallbac
   callbacks_ = &callbacks;
 }
 
+void IpTaggingFilter::applyTags(Http::RequestHeaderMap& headers,
+                                const std::vector<std::string>& tags) {
+  using HeaderAction = IpTaggingFilterConfig::HeaderAction;
+
+  const Http::LowerCaseString& headerName = config_->ip_tag_header();
```

Comment:
Yeah, I don't like it either, it's too "magic value" for my tastes.
Let me see if I can wrap things in a std::optional.

Reply:
I've changed it to an optional.
This means having to make the field in the class also an optional, which means it costs 4-8 bytes extra (it's in the config, it's shared, I think it's fine).

Rejected alternatives:
- using a raw pointer, because, well, raw-pointers are a good way to make mistakes in c++.
- passing an optional by value would incur a copy, which would be discarded in all code paths, because header_map::addCopy takes a const-reference.
- `optional<const Http::LowerCaseString&>` is not permitted in c++, because optional does not take references.

Let me know if you prefer this, or the old code. <3

Reply:
Could we try OptRef if that makes sense to you? It's a utility to do optional + ref

Reply:
Didn't know about that one. Let me give it a try. :)